pageserver/src/thread_mgr.rs

```diff
@@ -234,13 +234,13 @@ fn thread_wrapper<F>(
         Ok(Err(err)) => {
             if shutdown_process_on_error {
                 error!(
-                    "Shutting down: thread '{}' tenant_id: {:?}, timeline_id: {:?} exited with error: {:?}",
+                    "Shutting down: thread '{}' tenant_id: {:?}, timeline_id: {:?} exited with error: {:#}",
```

Comment:
Here and below. `{:#}` format does not produce backtraces so this changes visible behavior. Is it intentional?

Reply:
Yeah, it's intentional. Backtraces are still logged for internal error as specified in https://github.com/neondatabase/neon/pull/1713/files#diff-07e669dd036171046e713197ceed05a7a8294acfcd68f7ef9ce7a2d4ee056ec0R13-R14.

The idea is that if an error is expected, we can easily trace it by looking into the error message and cause.